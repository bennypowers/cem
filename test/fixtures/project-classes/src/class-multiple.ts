export class ClassA extends Event {
  aField: string;
  aMethod(): string {}
}

@customElement('class-b')
class ClassB extends LitElement {
  bField: string;
  @property() bProp: string;
  bMethod(): string {}
}
