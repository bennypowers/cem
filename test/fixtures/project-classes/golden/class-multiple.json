{
  "schemaVersion": "1.0.0",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "src/class-multiple.js",
      "declarations": [
        {
          "name": "ClassA",
          "superclass": {
            "name": "Event",
            "package": "global:"
          },
          "members": [
            {
              "name": "aField",
              "type": {
                "text": "string"
              },
              "kind": "field"
            },
            {
              "return": {
                "type": {
                  "text": "string"
                }
              },
              "name": "aMethod",
              "kind": "method"
            }
          ],
          "kind": "class"
        },
        {
          "name": "ClassB",
          "superclass": {
            "name": "LitElement",
            "package": "lit"
          },
          "members": [
            {
              "name": "bField",
              "type": {
                "text": "string"
              },
              "kind": "field"
            },
            {
              "name": "bProp",
              "type": {
                "text": "string"
              },
              "kind": "field",
              "attribute": "bprop"
            },
            {
              "return": {
                "type": {
                  "text": "string"
                }
              },
              "name": "bMethod",
              "kind": "method"
            }
          ],
          "kind": "class",
          "tagName": "class-b",
          "attributes": [
            {
              "name": "bprop",
              "type": {
                "text": "string"
              },
              "fieldName": "bProp"
            }
          ],
          "customElement": true
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "ClassA",
          "declaration": {
            "name": "ClassA",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/class-multiple.js"
          }
        },
        {
          "kind": "custom-element-definition",
          "name": "class-b",
          "declaration": {
            "name": "ClassB",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/class-multiple.js"
          }
        }
      ]
    }
  ]
}